mock/service/controller.go

```diff
@@ -159,6 +159,18 @@ func (s *service) ControllerUnpublishVolume(
 	req *csi.ControllerUnpublishVolumeRequest) (
 	*csi.ControllerUnpublishVolumeResponse, error) {
 
+	if len(req.VolumeId) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID cannot be empty")
+	}
+	if len(req.NodeId) == 0 {
+		// If node id is empty, no failure as per Spec
+		return &csi.ControllerUnpublishVolumeResponse{}, nil
```

Comment:
if the NodeId is empty do we want to return "success" immediately? Shouldn't it continue to run through the rest of the function

Reply:
if node id is empty, I think we need to just return success as per spec, because who knows on which node the operation should take place.

Reply:
as long as the spec says this, i'm ok.

Reply:
On second look, the spec for controllerunpublishvolume `If the value is unset, the SP MUST unpublish the volume from all nodes it is published to.`

We should have the mock driver implement this behavior.

Reply:
hmmm, ok I will take care of that then. Thanks David for digging deeper !!